### ISSUE TITLE
npm install without arguments shouldn't alter/save package.json

### ISSUE THREAD

**Issue by eamodio** [npm/npm#17010]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

npm 5 is saving `package.json` when `npm install` is run without any arguments. There shouldn't be a need to alter/save the `package.json` in this case.

Also see https://github.com/npm/npm/issues/16927

#### How can the CLI team reproduce the problem?

See https://github.com/npm/npm/issues/16927

### supporting information:

 - `npm -v` prints: `5.0.3`
 - `node -v` prints: `v7.10.0`
 - `npm config get registry` prints: `https://registry.npmjs.org/`
 - Windows, OS X/macOS, or Linux?: `Windows`
 - Network issues:
   - Geographic location where npm was run: `Northeastern US`
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by heinzdmx** [comment]
This appears to be the new default, run with --no-save if that is not what you want, see [release notes](https://github.com/npm/npm/releases/tag/v5.0.0)

**Comment by siddharthkp** [comment]
@eamodio This is a feature, not a bug 😄

You might want to check another file that it created: `package-lock.json`

**Comment by eamodio** [comment]
@heinzdmx will `--no-save` stop it from updating the `package-lock.json`?

@siddharthkp thanks, I'm aware of the that -- and I don't want to stop it from updating that -- it just has no reason to re-save `package.json` and because of https://github.com/npm/npm/issues/16927 there are consequences to that behavior.

**Comment by siddharthkp** [comment]
> will --no-save stop it from updating the package-lock.json?

Yes. It will also not touch package.json as well.

**Comment by eamodio** [comment]
Yeah - so that is *not* what I want. I want it to update the `package-lock.json` as it does now, but just stop re-saving the `package.json` for no reason.

**Comment by legodude17** [comment]
Could you do a diff of `package.json` before and after `npm i`?

**Comment by oliverjanik** [comment]
This is absolutely a bug. Who thought a breaking change like that would be a good idea? Where's the impact analysis? Is npm project run by bunch of kids? Silently breaking backwards compat for no good reason. 

This is probably breaking CI servers around the world.

**Comment by vkarponen** [comment]
How about you guys introduce an _npm sync_ command that synchronizes your workspace to match the package & package_lock files and never touches them (= essentially the same functionality as _npm install --no-save_).

Atm the documentation in https://docs.npmjs.com/files/package-locks seems false. This 
>"It is highly recommended you commit the generated package lock to source control: this will allow anyone else on your team, your deployments, your CI/continuous integration, and anyone else who runs npm install in your package source to get the exact same dependency tree that you were developing on." 

is not actually true since _npm install_ might change the contents of the package-lock file and therefore you don't end up with the exact same dependency tree. 

The danger with mixing installing and syncing is especially nasty on CI systems which might end up in a conflicted state after files from the version control get modified by "npm install".

**Comment by travi** [comment]
a couple more cases where i've found the new default to be troublesome:

* npm link - when unlinking, the linked dependency is removed from `package.json`. this prevents the previous ability to `npm unlink <package> && npm i` to get back to the previous real version
* attempting to set `save=false` in my config prevents proper updates to the lockfile as mentioned above. this is usually fine for me since i always install packages using the cli and include the proper flag, but when i use tools like [updtr](https://github.com/peerigon/updtr) that update the `package.json` and then run `npm install`, the corresponding lockfile updates are not applied.

i'd prefer a way to configure my default behavior to not update the `package.json` or `package-lock.json` from the cli unless i provide the proper flag, allowing temporarily testing a certain version w/o committing to it yet. but if it were possible to configure that way, i would still want changes applied to the `package-lock.json` when running `npm i` after updates are made to the `package.json`, either manually (rare for me) or by a tool.

**Comment by skaughtx0r** [comment]
This is especially annoying on windows.

If I just run `npm install`, it re-saves the package.json file with different line endings, so the file appears different to git. However there's no actual changes.

I think this is a bug.
